src/isomorphism.jl

```diff
@@ -0,0 +1,887 @@
+abstract type GraphMorphismProblemType end
+struct IsomorphismProblemType <: GraphMorphismProblemType end
+struct SubGraphIsomorphismProblemType <: GraphMorphismProblemType end
+struct InducedSubGraphIsomorphismProblemType <: GraphMorphismProblemType end
+
+const IsomorphismProblem = IsomorphismProblemType()
+const SubGraphIsomorphismProblem = SubGraphIsomorphismProblemType()
+const InducedSubGraphIsomorphismProblem = InducedSubGraphIsomorphismProblemType()
+
+
+"""
+    Box{T}
+A struct for passing references to functions
+"""
+mutable struct Box{T}
+    value::T
+end
+
+"""
+    has_induced_subgraphiso(g1, g2; vertex_relation=nothing, edge_relation=nothing, alg=:vf2)
+
+Returns `true` if the graph `g1` contains a vertex induced subgraph that is isomorphic to `g2`.
+
+### Optional Arguments
+- `vertex_relation`: A binary function that takes a vertex from `g1` and one from `g2`. An
+    isomorphism only exists if this function returns `true` for all matched vertices.
+- `edge_relation`: A binary function that takes an edge from `g1` and one from `g2`. An
+    isomorphism only exists if this function returns `true` for all matched edges.
+- `alg`: The algorithm that is used to find the induced subgraph isomorphism. Can be only
+    `:vf2` at the moment.
+
+### Examples
+```doctest.jl
+julia> has_induced_subgraphiso(CompleteGraph(5), CompleteGraph(4))
+true
+julia> has_induced_subgraphiso(CompleteGraph(5), CycleGraph(4))
+false
+
+julia> g1 = PathDiGraph(3); color1 = [1, 1, 1]
+julia> g2 = PathDiGraph(2); color2 = [1, 2]
+julia> color_rel(u, v) = (color1[u] == color2[v])
+julia> has_induced_subgraphiso(g1, g2)
+true
+julia> has_induced_subgraphiso(g1, g2, vertex_relation=color_rel)
+false
+```
+### See also 
+[`has_subgraphiso`](@ref), [`has_iso`](@ref), [`count_induced_subgraphiso`](@ref), [`all_induced_subgraphiso`](@ref)
+"""
+function has_induced_subgraphiso(g1::AbstractGraph, g2::AbstractGraph;
+                                 vertex_relation::Union{Nothing, Function}=nothing,
+                                 edge_relation::Union{Nothing, Function}=nothing,
+                                 alg=:vf2)::Bool
+    if alg == :vf2
+        boolbox = Box(false)
+        callback(vmap) = (boolbox.value = true; return false)
```

Comment:
I think that regular closures will catch the boolbox variable if you do it like this:

```julia
boolbox = false
callback(vmap) = (boolbox = true; return false)
vf2(callback, g1, g2, InducedSubGraphIsomorphismProblem;
                       vertex_relation=vertex_relation, edge_relation=edge_relation)
```

I just tested it with 
```julia
julia> function foo()
       a = true
       f(x) = a=false
       map(f, [1])
       return a
       end
foo (generic function with 1 method)

julia> foo()
false
```

and there is no need to use a Box because you are defining the callback after you define boolbox.

Reply:
You are right. I didn't know about `Ref` but as you said it is not necessary at all.